File: homeworks/Nigmatullin/gameHelper/gameHelper.py
f = open('dictionary.txt', 'r')

letters = ['а', 'б', 'а', 'к', 'а', 'б']

len_word = 5
flag = True
isPresent = False
lines = f.readlines()

for word in lines:
    word = word.strip()
    if len(word) != len_word:
        continue

    letters_temp = []
    letters_temp.extend(letters)

    flag = True

    for char in word:
        if letters_temp.count(char) > 0:
            letters_temp.remove(char)
        else:
            flag = False
            break

    if flag == True:
        isPresent = True
        print (word)

if isPresent == False:
    print('Нету')
f.close()

File: homeworks/Nigmatullin/gameHelper/gameHelper.cpp
#include <fstream>
#include <string>
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;
int main()
{
	setlocale(LC_ALL, "Russian");
	ifstream ifile("dictionary.txt");
	string word = "";
	int expectedNum = 5;

	vector<char> expectedChars;
	expectedChars.push_back('а');
	expectedChars.push_back('б');
	expectedChars.push_back('а');
	expectedChars.push_back('к');
	expectedChars.push_back('а');
	expectedChars.push_back('з');
	expectedChars.push_back('х');

	bool flag = true, isPresent = false;
	
	while (ifile >> word)
	{
		if (word.length() == expectedNum)
		{
			flag = true;
			vector<char> temp = expectedChars;
			for (int i = 0; i < word.length(); i++)
			{
				vector<char>::iterator tempIt = find(temp.begin(), temp.end(), word[i]);
				if (tempIt != temp.end())
				{
					temp.erase(tempIt);
				}
				else
				{
					flag = false;
					break;
				}
			}
			if (flag == true)
			{
				isPresent = true;
				cout << word << endl;
			}
		}
	}
	if (isPresent == false)
	{
		cout << "нету" << endl;
	}
	system("pause");
	return 0;
}

